test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenTransactSpecs.java

```diff
@@ -99,11 +106,166 @@ public static void main(String... args) {
 						uniqueTokenTxnWithSenderNotSigned(),
 						uniqueTokenTxnWithReceiverNotSigned(),
 						uniqueTokenTxnsAreAtomic(),
-						uniqueTokenDeletedTxn()
+						uniqueTokenDeletedTxn(),
+						cannotSendFungibleToDissociatedContractsOrAccounts(),
+						cannotGiveNftsToDissociatedContractsOrAccounts(),
+						recordsIncludeBothFungibleTokenChangesAndOwnershipChange(),
+						transferListsEnforceTokenTypeRestrictions(),
 				}
 		);
 	}
 
+	public HapiApiSpec transferListsEnforceTokenTypeRestrictions() {
+		final var theAccount = "anybody";
+		final var B_TOKEN = "non-fungible";
+		final var theKey = "multipurpose";
+		return defaultHapiSpec("TransferListsEnforceTokenTypeRestrictions")
+				.given(
+						newKeyNamed(theKey),
+						cryptoCreate(theAccount),
+						cryptoCreate(TOKEN_TREASURY),
+						tokenCreate(A_TOKEN)
+								.tokenType(TokenType.FUNGIBLE_COMMON)
+								.initialSupply(1000L)
+								.treasury(TOKEN_TREASURY),
+						tokenCreate(B_TOKEN)
+								.supplyKey(theKey)
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.initialSupply(0L)
+								.treasury(TOKEN_TREASURY)
+				).when(
+						mintToken(B_TOKEN, List.of(ByteString.copyFromUtf8("dark"))),
+						tokenAssociate(theAccount, List.of(A_TOKEN, B_TOKEN))
+				).then(
+						cryptoTransfer(
+								movingUnique(1, A_TOKEN).between(TOKEN_TREASURY, theAccount)
+						).hasKnownStatus(INVALID_NFT_ID),
+						cryptoTransfer(
+								moving(1, B_TOKEN).between(TOKEN_TREASURY, theAccount)
+						).hasKnownStatus(TOKEN_TRANSFERS_LIST_ONLY_USABLE_WITH_FUNGIBLE_COMMON)
```

Comment:
I think that `TOKEN_TRANSFERS_LIST_ONLY_USABLE_WITH_FUNGIBLE_COMMON ` might not be the best response code in this situation since technically speaking `Token Transfer List` is used with non-fungible uniques. NftTransfers are placed inside the `token-transfer-list`:

```
message TokenTransferList {
    TokenID token = 1; // The ID of the token
    repeated AccountAmount transfers = 2; // Applicable to tokens of type FUNGIBLE_COMMON. Multiple list of AccountAmounts, each of which has an account and amount
    repeated NftTransfer nftTransfers = 3; // Applicable to tokens of type NON_FUNGIBLE_UNIQUE. Multiple list of NftTransfers, each of which has a sender and receiver account, including the serial number of the NFT
}
```
 Maybe a better response code would be `INVALID_TOKEN_ID`

Reply:
Ah good point! Refined to `ACCOUNT_AMOUNT_TRANSFERS_ONLY_ALLOWED_FOR_FUNGIBLE_COMMON`.